LibCarla/source/carla/streaming/detail/tcp/ServerSession.cpp

```diff
@@ -80,8 +84,16 @@ namespace tcp {
         return;
       }
       if (_is_writing) {
-        log_debug("session", _session_id, ": connection too slow: message discarded");
-        return;
+        if (_server.IsSynchronousMode()) {
+          // wait until previous message has been sent
+          while (_is_writing) {
+            std::this_thread::yield();
+          }
```

Comment:
This could cause a dead-lock if you have as many or more sensors than threads. If you're very unlucky every session may end up spinning on this loop and there won't be any threads available to execute the callback.